### ISSUE TITLE
GroupedListView.grid should provide index for the entire list

### ISSUE THREAD

**Issue by adamcastleman** [quentin7b/flutter_grouped_listview#5]
First off, thank you for a grouped list package that supports gridview!

The itemBuilder of .grid() gives you the index of the item's place in each group, but you should be able to get the index of the entire list as well.

This would be necessary to create a "scroll to header" function and is available in most List/GridView objects and other grouped list packages.

Or if I'm missing something obvious...

**Comment by qk7b** [comment]
Hi ! 

Yeah that makes sense, If I understand well : giving 2 indexes might fill the need.

* One for the item in the group
* One for the group in the list of groups 

Is that the point ?

**Comment by adamcastleman** [comment]
Giving the index for the group in the list of groups would work for this use case, but I think it would be more flexible if you could get the index of the entire list independent of groupings.

If a list is being groupedBy a field, you should still be able to get each item’s index after that sorting.

If you had each item’s index in the entire list, you could easily call:

item.firstWhere((element) => element.itemId == index).first.itemName.

A scroll controller assigned to the list/grid could animateTo or jumpTo the position in the entire list. Actually, it would be best to jump to the header of that item.

It would also be good if you could assign an initial starting index of the list. Think a weekly list of tasks grouped by day, and you want to start the list on the current day.

**Comment by adamcastleman** [comment]
Yes, both would be perfect!

Also, an initial starting index would be useful.

Think a list of todo items grouped by day of week, and you want the list to begin on the current day.

**Comment by qk7b** [comment]
So given 

```dart
GroupedListView.grid({
    required Widget Function(
        BuildContext context, 
        int itemCountInGroup,
        int itemIndexInGroup, 
        I item
    ) gridItemBuilder
})
```

what you would like is

```dart
GroupedListView.grid({
    required Widget Function(
        BuildContext context, 
        int itemCountInGroup,
        int itemIndexInGroup, 
        I item, 
        {int positionInOriginalList}
    ) gridItemBuilder
})
```

This can be done with an `indexOf` in the builder but I'm scared of the performances here.
For each item, it will lookup its index.

**Comment by adamcastleman** [comment]
For smallish lists (probably majority of cases with groupings), performance should hold up.

is this being generated with the list/gridview. builder constructor? Items would be built lazily for super long lists.

**Comment by qk7b** [comment]
I came up with something better than `indexOf` which is not quite the best in case of duplicates. 
And also prevent any perf issue even when there is lazy loading or not.

**Comment by qk7b** [comment]
Fixed by #6
